qemu/tests/balloon_service.py

```diff
@@ -0,0 +1,234 @@
+            else:
+                logging.debug("Run Balloon Service in guest.")
+                try:
+                    run_balloon_service = params["run_balloon_service"] % drive_letter
+                    session.cmd(run_balloon_service)
+                except Exception:
+                    raise exceptions.TestError("Run balloon service failed !")
+        else:
+            logging.debug("Balloon Serice has already run.")
+        return drive_letter
+
+    def set_balloon_polling(device_path):
+        """
+        Set balloon polling in qmp.
+        cmd 'qom-set' only support in QMP.
+
+        :param device_path: balloon polling path.
+        """
+        set_balloon_property = params.get("set_balloon_property",
+                                          "guest-stats-polling-interval")
+        value = params.get("set_value", "2")
+        enable_polling_cmd = "qom-set path=%s,property=%s,value=%s"
+        try:
+            vm.monitor.send_args_cmd(enable_polling_cmd % (device_path,
+                                                           set_balloon_property,
+                                                           value))
+        except Exception:
+            raise exceptions.TestFail("Set balloon polling in qmp failed !")
+
+    def get_balloon_polling(device_path):
+        """
+        Get balloon polling in qmp.
+        cmd 'qom-get' only support in QMP.
+
+        :return resp: Reutn the output of 'qom-get'.
+        """
+        get_balloon_property = params.get("get_balloon_property", "guest-stats")
+        try:
+            resp = vm.monitor.cmd_qmp('qom-get', {'path': device_path, 'property': get_balloon_property})
+            return resp
+        except Exception:
+            raise exceptions.TestFail("Get balloon polling stats in qmp failed !")
+
+    def get_memory_status(session):
+        """
+        Get Memory status inside guest.
+
+        :param session: VM session.
+        :return: the free memory size inside guest.
+        :rtype: int
+        """
+        logging.debug("Get free memory in guest.")
+        free_mem_cmd = params['free_mem_cmd']
+        try:
+            # In Windows guest we get the free memory for memory compare
+            memory = session.cmd_output(free_mem_cmd)
+            memory = int(re.findall(r"\d+", memory)[0])
+            memory *= 1024 ** -1
+        except Exception, emsg:
+            logging.error(emsg)
+            return 0
+        return memory
+
+    def memory_check(session, get_polling_output, keyname):
+        """
+        Check memory status.
+
+        :param session: VM session.
+        :param get_polling_output: output of get polling in qmp.
+        :param keyname: key name of the output of the 'qom-get' property.
+        """
+        error_context.context("Check whether memory status as expected",
+                              logging.info)
+        ratio = params.get("ratio", "0.1")
```

Comment:
ratio is float type as default.
Could you help provide more details? I have not met such error in my testing.
Not sure why there is type error.
Thanks.